### PR TITLE
Context Overlays - new render stuff. PREPARE FOR EDGE HIGHLIGHTING

### PR DESCRIPTION
**Pull request by zfox23**
DO NOT TEST

this PR is crazy.

### CODE REVIEW

**Comment by marko8904** [comment]
build this please

**Comment by marko8904** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2028/HighFidelity-Beta-PR11057-8a7561d61ba30391147691349c3ecab9416cf46a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows2016/2028/HighFidelity-Beta-PR11057-8a7561d61ba30391147691349c3ecab9416cf46a.exe

File: interface/src/ui/overlays/ContextOverlayInterface.cpp [hunk 1]
```diff
@@ -104,17 +118,17 @@ bool ContextOverlayInterface::createOrDestroyContextOverlay(const EntityItemID&
...
-                PickRay pickRay(cameraPosition, direction);
-                _bbOverlay->setIgnoreRayIntersection(false);
-                auto result = qApp->getOverlays().findRayIntersection(pickRay);
-                _bbOverlay->setIgnoreRayIntersection(true);
-                contextOverlayPosition = result.intersection - direction * CONTEXT_OVERLAY_FAR_OFFSET;
+                glm::vec3 direction = glm::normalize(entityPosition - cameraPosition);
+                float distance;
+                BoxFace face;
+                glm::vec3 normal;
+                bool intersectionExists = boundingBox.findRayIntersection(cameraPosition, direction, distance, face, normal);
```
Comment (davidkelly) [inline interface/src/ui/overlays/ContextOverlayInterface.cpp]: This intersectionExists does not get used.  Maybe just remove the bool intersectionExists, but keep the call?

File: interface/src/ui/overlays/ContextOverlayInterface.cpp [hunk 2]
```diff
@@ -104,17 +118,17 @@ bool ContextOverlayInterface::createOrDestroyContextOverlay(const EntityItemID&
...
+                contextOverlayPosition = (cameraPosition + direction * distance) - direction * CONTEXT_OVERLAY_FAR_OFFSET;
```
Comment (manapa01) [inline interface/src/ui/overlays/ContextOverlayInterface.cpp]: "distance" is used here, but above it is never initialized. Why is that?

Comment (davidkelly) [inline interface/src/ui/overlays/ContextOverlayInterface.cpp]: it is filled in by findRayIntersection -- we just should not have removed that call (just not saved the return value as a variable)

**Comment by davidkelly** [comment]
So far the only issue I see is you don't get the highlight when you are holding the object you are lasering over.  I often (maybe not always) see the highlight for a very brief moment, but generally there's just the (i) off to the side.  Mouseover in desktop is great.  Personally I'd want it to work the same as in HMD, and bring up the (i) after some time, but maybe there's a reason to still trigger it off the right click.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2186/HighFidelity-Beta-PR11057-762b4098143b4fa68cf10e16e232ca5229fe6a7f.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2186/HighFidelity-Beta-PR11057-762b4098143b4fa68cf10e16e232ca5229fe6a7f.exe

**Comment by zfox23** [comment]
I noticed this too, and I'm concerned this a rendering issue. You'll notice that, if you hover over something and see the bounding box, then look away, then look back (hovering the whole time), the box will disappear...Seems related to the issue you saw. I didn't have time to fix it last week, and, like I said, it might be a rendering thing specific to bounding boxes (and thus may not be present when we switch to edge highlighting).

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/2187/HighFidelity-Beta-PR11057-588afaaa85861ea2f0ca70446cbc0d510d70c6a6.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/2187/HighFidelity-Beta-PR11057-588afaaa85861ea2f0ca70446cbc0d510d70c6a6.exe

**Pull request merged**